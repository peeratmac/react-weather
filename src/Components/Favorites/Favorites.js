import React, { Component } from 'react';
import './Favorites.scss';
import { fetchUsingStationID, fetchPopularCities } from '../../apiCalls';
import WeatherCard from '../../Components/WeatherCard/WeatherCard';
import { connect } from 'react-redux';
export class Favorites extends Component {
  componentDidMount() {
    console.log('yep');
    console.log(this.props.stationIDs);
  }

  render() {
    console.log('LIST-1', this.props);
    const { weatherInfo } = this.props;
    return weatherInfo.map((city, i) => {
      return (
        <div className='card'>
          <h1>City: {city.title}</h1>
          <h1>
            <span>
              Current Condition:{' '}
              {city.consolidated_weather[0].weather_state_name}
            </span>
          </h1>
          <h2>
            <span>
              Min Temp (Degree Celsius): {city.consolidated_weather[0].min_temp}
            </span>
          </h2>
          <h2>
            <span>
              Max Temp (Degree Celsius): {city.consolidated_weather[0].max_temp}
            </span>
          </h2>
          <h2>
            <span>Humidity (%): {city.consolidated_weather[0].humidity}</span>
          </h2>
        </div>
      );
    });
    return (
      <div>
        <h1>LIST 1 PAGE</h1>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  cities: state.cities,
  stationIDs: state.stationIDs,
  weatherInfo: state.weatherInformation
});

export default connect(
  mapStateToProps,
  null
)(Favorites);
