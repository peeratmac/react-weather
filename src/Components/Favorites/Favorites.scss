* {
  box-sizing: border-box;
  font-family: 'Rubik', sans-serif;
}

$base-weather: #38ada9;
$current-condition: #ffffff;
$min-temperature: #6a89cc;
$max-temperature: #fa983a;
$humidity-label: #7f8fa6;
$humidity: #8e8e91;

.forecast-card {
  border: 10px solid $base-weather;
  border-radius: 25px;
  margin: 10px;
  text-align: center;
  padding: 20px;
}

.current-condition {
  background-color: $current-condition;
  padding: 5px;
  border-radius: 5px;
}

.min-temp-label,
.max-temp-label {
  background-color: $base-weather;
  padding: 5px;
  border-radius: 5px;
}

.min-temp {
  background-color: $min-temperature;
  padding: 5px;
  border-radius: 5px;
}

.max-temp {
  background-color: $max-temperature;
  padding: 5px;
  border-radius: 5px;
}

.humidity-label {
  background-color: $humidity-label;
  padding: 5px;
  border-radius: 5px;
}

.humidity {
  background-color: $humidity;
  padding: 5px;
  border-radius: 5px;
}
