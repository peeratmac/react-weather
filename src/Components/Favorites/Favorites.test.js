import React from 'react';
import { shallow } from 'enzyme';
import { Favorites } from './Favorites';

describe('FAVORITES', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Favorites cities={''} stationIDs={[]} weatherInfo={[]} />
    );
  });

  it('should match the Favorites Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
