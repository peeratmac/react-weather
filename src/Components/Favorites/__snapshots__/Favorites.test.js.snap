// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAVORITES should match the Favorites Snapshot 1`] = `
<div>
  <h1>
    LIST 1 PAGE
  </h1>
</div>
`;
